src/test/java/seedu/address/TestApp.java

```diff
@@ -84,18 +119,36 @@ public EntryBook readStorageAddressBook() {
     }
 
     /**
-     * Returns the file path of the storage file.
+     * Returns the file path of the storage file for the list entry book.
      */
-    public Path getStorageSaveLocation() {
-        return storage.getAddressBookFilePath();
+    public Path getListEntryBookStorageSaveLocation() {
+        return storage.getListEntryBookFilePath();
+    }
+
+    /**
+     * Returns the file path of the storage file for the archives entry book.
+     */
+    public Path getArchivesEntryBookStorageSaveLocation() {
+        return storage.getArchivesEntryBookFilePath();
     }
 
     /**
      * Returns a defensive copy of the model.
      */
     public Model getModel() {
-        Model copy = new ModelManager(model.getListEntryBook(), new UserPrefs(), new StorageStub());
+        Model copy = new ModelManager(model.getListEntryBook(), model.getArchivesEntryBook(),
+            new UserPrefs(), new StorageStub());
         copy.setContext(model.getContext());
+        // This is needed to ensure the copy also has the correct entrybook displayed
+        switch (copy.getContext()) {
+        case CONTEXT_LIST:
```

Comment:
Context and the displayed `EntryBook` might not always be in 1-1 relation, for example when viewing all links in the current article or opening a feed.

Reply:
Maybe we can use `Model#clone`?

Reply:
> Maybe we can use `Model#clone`?

I actually did this in my previous PR for context-swinging, then reverted it.
```Model#clone``` clones the storage component as well, so the "clone" is not a completely separate clone.